src/utils.cpp

```diff
@@ -0,0 +1,38 @@
+#include "utils.h"
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+#include <net/if.h>
+#include <netinet/in.h>
+
+namespace zeek {
+std::vector<std::string> getHostIPAddrs() {
+  std::vector<std::string> ip_addrs;
+
+  struct ifaddrs *ifAddrStruct = NULL;
+  struct ifaddrs *ifa = NULL;
+  void *tmpAddrPtr = NULL;
+
+  getifaddrs(&ifAddrStruct);
+
+  for (ifa = ifAddrStruct; ifa != NULL; ifa = ifa->ifa_next) {
+    if (!ifa->ifa_addr || (ifa->ifa_flags & IFF_LOOPBACK)) {
```

Comment:
Are we sure there is no a way for loopback addresses ending up in events? I am wondering whether we should filter these here at all since from the outside a loopback address would not be obvious.

Reply:
This is specific to my case, where I am creating a map of IP address -> Host ID in Zeek. And loopback addresses are of no use for that case. So I am not sending them to Zeek. I think it is okay to filter them here for that case.

Reply:
I am thinking of cases where connections to loopback addresses might lead to events somewhere, and us that dropping this address here. This would make it hard for users of the data to reliably filter events.

How about just adding loopback addresses as well? Instead of emitting a vector of addresses we could emit a vector of tuples (address, is loopback?).

Reply:
After talking with @Wajihulhassan offline I agree that reporting loopback address is not that useful since we do not have the equivalent of zeek running on the agents. Dropping.